torch/csrc/distributed/autograd/context/DistAutogradContext.cpp

```diff
@@ -12,6 +12,17 @@ int64_t DistAutogradContext::context_id() const {
   return context_id_;
 }
 
+void DistAutogradContext::addSendFunction(
+    std::shared_ptr<SendRpcBackwards> func) {
+  std::lock_guard<std::mutex> guard(lock_);
+  sendAutogradFunctions_.push_back(func);
+}
+
+std::vector<std::shared_ptr<SendRpcBackwards>> DistAutogradContext::
```

Comment:
Should this be a map, as recv need to use its ID to find the corresponding send??

Reply:
We can modify this in follow up PRs to a map, right now we don't have an ID to put in this map.